### PR TITLE
Fix sampling and Kafka reporter conflicting.

### PR DESCRIPTION
**Pull request by wu-sheng**
@peng-yongsheng reported this conflict. As they used to extend the same core service, when both of them are active, which could happen, they can't work together.

But, from the design perspective, I notice, ignore plugin actually is just a sampling service. It used `ContextManagerExtendService` extension, just because only sampling service API doesn't provide the operation name.

In this PR, I changed the sampling service API, and move `TraceIgnoreExtendService` to extend it. So conflict goes away.

### CODE REVIEW

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java [hunk 1]
```diff
@@ -438,20 +434,7 @@ private void finish() {
...
             if (isFinishedInMainThread && (!isRunningInAsyncMode || asyncSpanCounter == 0)) {
                 TraceSegment finishedSegment = segment.finish(isLimitMechanismWorking());
-                /*
-                 * Recheck the segment if the segment contains only one span.
-                 * Because in the runtime, can't sure this segment is part of distributed trace.
-                 *
-                 * @see {@link #createSpan(String, long, boolean)}
-                 */
-                if (!segment.hasRef() && segment.isSingleSpanSegment()) {
-                    if (!SAMPLING_SERVICE.trySampling()) {
```
[the remark this answers is not in the document]
Comment (wu-sheng) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java:448]: Removed.

**Review by kezhenxu94 — APPROVED** [review]

**Comment by liuzhiguo630** [comment]
This commit makes TraceIgnoreExtendService overrides SamplingService, so sample count or trace prefix only pick one?

**Comment by wu-sheng** [comment]
It used to be in either, not both. This is fixing other bugs. `TraceIgnoreExtendService` technically is a sampling mechanism, but base on rules.

**Comment by liuzhiguo630** [comment]
I understand what's your means.And we are already cherry-picked this commit to our 8.1.0 to solve conflict, I mean for functionaly, filter trace by operation name and limit by count should be both.
I'm trying write a new SamplingService to override it, but I still hope these two function could be original.

**Comment by wu-sheng** [comment]
> I'm trying write a new SamplingService to override it, but I still hope these two function could be original.

You could simply contribute this to the ignore plugin, which could just call `super#doSampling`. We are welcome to contribute to this feature.

**Pull request merged**